cpp/turbodbc_arrow/Library/src/arrow_result_set.cpp

```diff
@@ -125,14 +130,26 @@ Status append_to_date_builder(size_t rows_in_batch, std::unique_ptr<ArrayBuilder
     return Status::OK();
 }
 
-Status append_to_string_builder(size_t rows_in_batch, std::unique_ptr<ArrayBuilder> const& builder, cpp_odbc::multi_value_buffer const& input_buffer, uint8_t*) {
-    auto typed_builder = static_cast<StringBuilder*>(builder.get());
-    for (std::size_t j = 0; j != rows_in_batch; ++j) {
-        auto const element = input_buffer[j];
-        if (element.indicator == SQL_NULL_DATA) {
-            ARROW_RETURN_NOT_OK(typed_builder->AppendNull());
-        } else {
-            ARROW_RETURN_NOT_OK(typed_builder->Append(element.data_pointer, element.indicator));
+Status append_to_string_builder(size_t rows_in_batch, std::unique_ptr<ArrayBuilder> const& builder, cpp_odbc::multi_value_buffer const& input_buffer, uint8_t*, bool strings_as_dictionary) {
+    if (strings_as_dictionary) {
+        auto typed_builder = static_cast<StringDictionaryBuilder*>(builder.get());
```

Comment:
How about extracting the blocks of the `if` blocks to a template function in the anonymous namespace? It is the same code, the only difference is the type of the `typed_builder`.